plugin/pkg/auth/authorizer/node/graph.go

```diff
@@ -248,6 +248,30 @@ func (g *Graph) deleteEdges_locked(fromType, toType vertexType, toNamespace, toN
 	}
 }
 
+// A fastpath for recomputeDestinationIndex_locked for "adding edge case".
+func (g *Graph) addEdgeToDestinationIndex_locked(e graph.Edge) {
+	n := e.From()
+	// don't maintain indices for nodes with few edges
+	edgeCount := g.graph.Degree(n)
+	if edgeCount < g.destinationEdgeThreshold {
+		delete(g.destinationEdgeIndex, n.ID())
+		return
+	}
```

Comment:
this duplicates code that already exists inside recomputeDestinationIndex_locked. would this be simpler as:

```go
func (g *Graph) addEdgeToDestinationIndex_locked(e graph.Edge) {
	n := e.From()
	index := g.destinationEdgeIndex[n.ID()]
	if index == nil {
		// There is no index, use the full index computation method
		g.recomputeDestinationIndex_locked(n)
		return
	}
	// fast-add the new edge to an existing index
	if destinationEdge, ok := e.(*destinationEdge); ok {
		index.mark(destinationEdge.DestinationID())
	}
}
```

Reply:
Good idea. Thanks!